CMakeLists.txt

```
@@ -345,6 +345,17 @@ if(IWYU_FOUND)
   add_custom_target(checkHEADERS DEPENDS $ENV{ALL_HEADER_RULES})
 endif(IWYU_FOUND)
 
+if($ENV{FAIRROOT_FORMAT_BASE_COMMIT})
+  set(FORMAT_BASE_COMMIT $ENV{FAIRROOT_FORMAT_BASE_COMMIT})
+else()
+  set(FORMAT_BASE_COMMIT HEAD)
+endif()
+
+add_custom_target(ClangFormatCheck
+  COMMAND ${CMAKE_SOURCE_DIR}/check-clang-format.sh ${FORMAT_BASE_COMMIT}
```

```suggestion
add_custom_target(ClangFormatCheck
  COMMAND ${CMAKE_SOURCE_DIR}/check-clang-format.sh \$FORMAT_BASE_COMMIT
```

not sure if this is the proper escaping, but you want the environment to be read at exec time of the target, not at cmake configure time.

---

probably best to move the whole logic inside the `check-clang-format.sh` script

---

done.